### PR TITLE
net-misc/batctl: Version bump to 2020.2, clean up

### PR DESCRIPTION
**Pull request by HarrySmallbones**
Signed-off-by: Harry Smallbones <info@harrysmallbones.co.uk>
Package-Manager: Portage-2.3.99, Repoman-2.3.22

### CODE REVIEW

File: net-misc/batctl/batctl-2020.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+SRC_URI="https://downloads.open-mesh.org/batman/releases/batman-adv-${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="dev-libs/libnl:3"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
```
Comment (DarthGandalf) [inline net-misc/batctl/batctl-2020.2.ebuild:19]: pkgconfig should be in BDEPEND

File: net-misc/batctl/batctl-2020.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		ewarn "batctl requires batman-adv kernel support"
+	fi
+}
+
+src_compile() {
+	emake CC="$(tc-getCC)" V=1 REVISION=gentoo-"${PVR}"
+}
+
+src_install() {
+	emake DESTDIR="${D}" PREFIX="${EPREFIX}"/usr install
```
Comment (thesamesam) [inline net-misc/batctl/batctl-2020.2.ebuild:32]: Maybe quote the whole thing again, but this isn't compulsory.

File: net-misc/batctl/batctl-2020.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+EAPI=7
+
+inherit linux-info toolchain-funcs
+
+DESCRIPTION="BATMAN advanced control and management tool"
+HOMEPAGE="https://www.open-mesh.org/"
+SRC_URI="https://downloads.open-mesh.org/batman/releases/batman-adv-${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2"
```
Comment (thesamesam) [inline net-misc/batctl/batctl-2020.2.ebuild:12]: I noticed there was a LICENSES folder with 'deprecated' and 'preferred', and that the SPDX-License-Identifier header was used on files in the project.

Here's what turned up:
```
$ grep -rsin "SPDX-License-Identifier: MIT" .                                                                                                                                            
./batman_adv.h:1:/* SPDX-License-Identifier: MIT */
./LICENSES/preferred/MIT:7:    SPDX-License-Identifier: MIT
./list.h:1:/* SPDX-License-Identifier: MIT */
```

```
$ grep -rsin "SPDX-License-Identifier: ISC" .
./LICENSES/deprecated/ISC:7:    SPDX-License-Identifier: ISC
./genl.c:1:// SPDX-License-Identifier: ISC
```

File: net-misc/batctl/batctl-2020.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="dev-libs/libnl:3"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+pkg_setup() {
+	if ! ( linux_config_exists && linux_chkconfig_present BATMAN_ADV ); then
```
Comment (thesamesam) [inline net-misc/batctl/batctl-2020.2.ebuild:22]: We don't _need_ to use a subshell here. So I'd prefer to use the standard syntax for ebuilds:
```suggestion
	if ! [[ linux_config_exists && linux_chkconfig_present BATMAN_ADV ]]; then
```
or similar?

File: net-misc/batctl/batctl-2020.2.ebuild (hunk 1 above)
Comment (HarrySmallbones) [reply net-misc/batctl/batctl-2020.2.ebuild]: Thanks, updated.

File: net-misc/batctl/batctl-2020.1.ebuild [hunk 5]
```diff
@@ -19,10 +19,8 @@ DEPEND="${RDEPEND}
 	virtual/pkgconfig"
```
[the remark this answers is not in the document]
Comment (HarrySmallbones) [reply net-misc/batctl/batctl-2020.1.ebuild]: Thanks - I've moved pkgconfig into BDEPEND.

File: net-misc/batctl/batctl-2020.2.ebuild [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+		ewarn "batctl requires batman-adv kernel support"
+	fi
+}
+
+src_compile() {
+	emake CC="$(tc-getCC)" V=1 REVISION=gentoo-"${PVR}"
```
[the remark this answers is not in the document]
Comment (HarrySmallbones) [reply net-misc/batctl/batctl-2020.2.ebuild]: Updated

File: net-misc/batctl/batctl-2020.2.ebuild (hunk 2 above)
Comment (HarrySmallbones) [reply net-misc/batctl/batctl-2020.2.ebuild]: Agreed, looks neater. Updated.

File: net-misc/batctl/batctl-2020.2.ebuild (hunk 3 above)
Comment (HarrySmallbones) [reply net-misc/batctl/batctl-2020.2.ebuild]: Added the two other licences.

File: net-misc/batctl/batctl-2020.2.ebuild (hunk 4 above)
Comment (HarrySmallbones) [reply net-misc/batctl/batctl-2020.2.ebuild]: Updated

**Pull request closed without merging**